engine/common/follower/pending_tree/pending_tree.go

```diff
@@ -9,40 +9,17 @@ import (
 	"github.com/onflow/flow-go/module/mempool"
 )
 
-// CertifiedBlock holds a certified block, it consists of a block and a QC which proves validity of block (QC.BlockID = Block.ID())
-// This is used to compactly store and transport block and certifying QC in one structure.
-type CertifiedBlock struct {
-	Block *flow.Block
-	QC    *flow.QuorumCertificate
-}
-
-// ID returns unique identifier for the certified block
-// To avoid computation we use value from the QC
-func (b *CertifiedBlock) ID() flow.Identifier {
-	return b.QC.BlockID
-}
-
-// View returns view where the block was produced.
-func (b *CertifiedBlock) View() uint64 {
-	return b.QC.View
-}
-
-// Height returns height of the block.
-func (b *CertifiedBlock) Height() uint64 {
-	return b.Block.Header.Height
-}
-
 // PendingBlockVertex wraps a block proposal to implement forest.Vertex
 // so the proposal can be stored in forest.LevelledForest
 type PendingBlockVertex struct {
-	CertifiedBlock
+	flow.CertifiedBlock
 	connectedToFinalized bool
 }
 
 var _ forest.Vertex = (*PendingBlockVertex)(nil)
 
 // NewVertex creates new vertex while performing a sanity check of data correctness.
-func NewVertex(certifiedBlock CertifiedBlock, connectedToFinalized bool) (*PendingBlockVertex, error) {
+func NewVertex(certifiedBlock flow.CertifiedBlock, connectedToFinalized bool) (*PendingBlockVertex, error) {
 	if certifiedBlock.Block.Header.View != certifiedBlock.QC.View {
```

Comment:
since `CertifiedBlock` has been moved to `flow` I think we really need to be careful with holding invariant that:
`block.Header.View == qc.View && block.ID() == qc.BlockID`, this was previously implemented in `NewVertex` but now I think we would benefit having a constructor for `CertifiedBlock` that ensures we create a valid `CertifiedBlock`

Reply:
Good idea to enforce the necessary consistency requirements in a constructor! Overall, it seemed doable (most of the methods that are constructing CertifiedBlocks already had error returns and for the few that didn't it was straight forward to add an error return). 

implemented in https://github.com/onflow/flow-go/pull/4155/commits/19837298cd5996bca2c338fd2b7845e15c6d9ead

Hopefully the code passes the tests 🤞